transport/server.cc

```diff
@@ -677,7 +677,7 @@ future<> cql_server::connection::process_request() {
                 ? get_units(_server._memory_available, mem_estimate, shedding_timeout).then_wrapped([this, length = f.length] (auto f) {
                     try {
                         return make_ready_future<semaphore_units<>>(f.get0());
-                    } catch (semaphore_timed_out sto) {
+                    } catch (semaphore_timed_out& sto) {
```

Comment:
Is the std::move(sto) below legal/sensible?
In the captures, you can just capture &sto. In the "return", I think we need to copy.

Reply:
I think you're right. Since the reference to the exception object isn't const, the user can do anything valid to change this object, including moving things out of it. The user code doesn't "own" this object (it doesn't need to free it), but it may change it.

The std::move in the capture is unnecessary, you could have captured by reference since it was already a reference, but not an error either. So your PR looks good.